code/controllers/subsystem/shuttle.dm

```diff
@@ -217,6 +220,93 @@ SUBSYSTEM_DEF(shuttle)
+		return
+
+	var/time_to_recall = rand(600, 1500)
+	recall_timer_id = addtimer(CALLBACK(src, .proc/recall_illegitimate, emergency.timer), time_to_recall, TIMER_STOPPABLE)
+	if(!annoyed_admiral_message)
+		annoyed_admiral_message = pick("Do you know how expensive these stations are?","Stop wasting my time.","I was sleeping, thanks a lot.","Stand and fight you cowards!","You knew the risks coming in.","Stop being paranoid.","Whatever's broken just build a new one.","No.", "<i>null</i>","<i>Error: No comment given.</i>", "It's a good day to die!")
+
+	message_admins("Shuttle will recall automatically. (<A HREF='?_src_=holder;stop_autorecall=\ref[usr]'>STOP AUTORECALL</A>) (<A HREF='?_src_=holder;set_annoyed_admiral_message=1'>SET ANNOYED ADMIRAL MESSAGE</A>)")
+
+/datum/controller/subsystem/shuttle/proc/recall_illegitimate(old_timer)
+	if(emergency.mode != SHUTTLE_CALL || emergency.timer != old_timer)
+		return
+
+	emergency.cancel(/area/centcom)
+	msglog_admins("Shuttle has been automatically recalled.")
+
+	sleep(50)
+	var/intercepttext = "<FONT size = 3><b>NanoTrasen Update</b>: Request For Shuttle.</FONT><HR>\
+						To whom it may concern:<BR><BR>\
+						We have taken note of the situation upon [station_name()] and have come to the \
+						conclusion that it does not warrant the abandonment of the station.<BR>\
+						If you do not agree with our opinion we suggest that you open a direct \
+						line with us and explain the nature of your crisis.<BR><BR>\
+						<i>This message has been automatically generated based upon readings from long \
+						range diagnostic tools. To assure the quality of your request every finalized report \
+						is reviewed by an on-call rear admiral.<BR>\
+						<b>Rear Admiral's Notes:</b> \
+						[annoyed_admiral_message]"
+	annoyed_admiral_message = null
+	print_command_report(intercepttext, announce=TRUE)
+
+/datum/controller/subsystem/shuttle/proc/station_in_trouble() //Checks are sorted in rough order of processing cost
+	. = FALSE
+	msglog_admins("Emergency shuttle legitimacy checks follow:")
+
+	listclearnulls(SSshuttle.horrible_things)
+	var/num_things = SSshuttle.horrible_things.len
+	msglog_admins("[num_things] horrible things on station[num_things ? ":[english_list(SSshuttle.horrible_things)]" : ""]")
+
+	if(num_things)
+		. = TRUE
+
+	msglog_admins("Round length is [round(world.time / 600)] minutes, boredom threshold is set to [config.shuttle_boredom_check] minutes.")
+	if(world.time >= (config.shuttle_boredom_check * 600)) //Extended mercy and/or boring/ineffective antags
+		. = TRUE
+
+	var/list/living_crew_bodies = list()
+	var/list/living_crew_minds	= list()
+	for(var/mob/Player in GLOB.mob_list)
+		if(Player.mind && Player.stat != DEAD && !isnewplayer(Player) &&!isbrain(Player))
+			living_crew_bodies += Player
+			living_crew_minds += Player.mind
+
+	var/living_ratio
+	if(!GLOB.joined_player_list.len)
+		living_ratio = INFINITY
```

Comment:
Shouldn't you just skip the below code if there aren't any players here?
As is the admins would get a message saying `1x10^29% of crew are alive`